microcks.go

```diff
@@ -91,44 +97,76 @@ func WithSecondaryArtifact(artifactFilePath string) testcontainers.CustomizeRequ
 // WithArtifact provides paths to artifacts that will be imported within the Microcks container.
 // Once it will be started and healthy.
 func WithArtifact(artifactFilePath string, main bool) testcontainers.CustomizeRequestOption {
-	return func(req *testcontainers.GenericContainerRequest) {
+	return func(req *testcontainers.GenericContainerRequest) error {
 		hooks := testcontainers.ContainerLifecycleHooks{
 			PostReadies: []testcontainers.ContainerHook{
 				importArtifactHook(artifactFilePath, main),
 			},
 		}
 		req.LifecycleHooks = append(req.LifecycleHooks, hooks)
+
+		return nil
 	}
 }
 
-// WithNetwork allows to add a custom network
+// WithNetwork allows to add a custom network.
+// Deprecated: Use network.WithNetwork from testcontainers instead.
 func WithNetwork(networkName string) testcontainers.CustomizeRequestOption {
-	return func(req *testcontainers.GenericContainerRequest) {
+	return func(req *testcontainers.GenericContainerRequest) error {
 		req.Networks = append(req.Networks, networkName)
+		return nil
 	}
 }
 
-// WithNetworkAlias allows to add a custom network alias for a specific network
+// WithNetworkAlias allows to add a custom network alias for a specific network.
+// Deprecated: Use network.WithNetwork from testcontainers instead.
 func WithNetworkAlias(networkName, networkAlias string) testcontainers.CustomizeRequestOption {
-	return func(req *testcontainers.GenericContainerRequest) {
+	return func(req *testcontainers.GenericContainerRequest) error {
 		if req.NetworkAliases == nil {
 			req.NetworkAliases = make(map[string][]string)
 		}
 		req.NetworkAliases[networkName] = []string{networkAlias}
+
+		return nil
 	}
 }
 
-// WithEnv allows to add an environment variable
+// WithEnv allows to add an environment variable.
 func WithEnv(key, value string) testcontainers.CustomizeRequestOption {
-	return func(req *testcontainers.GenericContainerRequest) {
+	return func(req *testcontainers.GenericContainerRequest) error {
 		if req.Env == nil {
 			req.Env = make(map[string]string)
 		}
 		req.Env[key] = value
+
+		return nil
+	}
+}
+
+// WithHostAccessPorts allows to set the host access ports.
+func WithHostAccessPorts(hostAccessPorts []int) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.HostAccessPorts = hostAccessPorts
+
+		return nil
+	}
+}
+
+// WithSecret allows to add a new secret.
+func WithSecret(s client.Secret) testcontainers.CustomizeRequestOption {
```

Comment:
We're using the `client.Secret` interface here. Shouldn't we remove the local interface type definition above?

Reply:
Clearly, done ✅